pkg/apis/networking/types.go

```diff
@@ -362,3 +405,39 @@ type IngressBackend struct {
 	// Specifies the port of the referenced service.
 	ServicePort intstr.IntOrString
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IngressClass represents the class of the Ingress, referenced by the Ingress
+// Spec.
+type IngressClass struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	// Controller is responsible for handling this class. This should be
+	// specified as a domain-prefixed path, e.g. "acme.io/ingress-controller".
+	// This allows for different "flavors" that are controlled by the same
+	// controller. For example, you may have different Parameters for the same
+	// implementing controller.
+	Controller string
+
+	// Parameters is a link to a custom resource configuration for the
+	// controller. This is optional if the controller does not require extra
+	// parameters.
+	// +optional
+	Parameters *api.ObjectReference
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IngressClassList is a collection of IngressClasses.
+type IngressClassList struct {
```

Comment:
Do you have a use-case in mind for a collection of IngressClasses? I don't recall stacking IngressClasses coming up before, but I may have missed a meeting.

Reply:
I think this is just a required type to support listing resources as part of API calls. No intent on some kind of meaningful list resource, just enough that if someone has multiple IngressClasses in a cluster they should be able to list them. I could be wrong on this one, I've just always seen the corresponding `List` type for k8s resources.